### ISSUE TITLE
[wish] Saving the ingame settings

### ISSUE THREAD

**Issue by Spikeone** [Return-To-The-Roots/s25client#55]
The settings, which the player can change in the game, should be saved, maybe in an .ini file. Some settings for exmple are the distribution of goods or order of transportet goods etc.

https://bugs.launchpad.net/s25rttr/+bug/367477

**Comment by Spikeone** [comment]
https://bugs.launchpad.net/s25rttr/+bug/389952

Im original S2 konnte man die Positionen der Fenster wie z.B. das Militärfenster oder das Statistikfenster an einem Ort platzieren und nach dem Laden waren sie dann auch wieder dort.
Auch wenn man ein Fenster jetzt schliesst und es wieder öffnet, dann erscheint es immer in der Mitte oder oben (bei einigen Fenstern).

Ich fände es noch nett, wenn das noch angepasst würde. Das erspart etwas Zeit pro Spiel und macht dann auch mehr spass ^^

**Comment by Spikeone** [comment]
https://bugs.launchpad.net/s25rttr/+bug/1066384

Example of an config file for Unlimited play and Multiplayer

https://launchpadlibrarian.net/119680070/s2rttr.cfg-example-suggestion.cfg

**Comment by Spikeone** [comment]
just saying, saving some settings like tool settings, military settings etc. doesn't make sense at all as those are always dependent on addon settings and current game stage

**Comment by JonathanSteinbuch** [comment]
I agree that it doesn't make sense to just save tool settings and military settings from the state at the end of the game. But as user-definable presets for the start of the game they would make sense, in my opinion.

**Comment by Spikeone** [comment]
> 
> 
> I agree that it doesn't make sense to just save tool settings and military settings from the state at the end of the game. But as user-definable presets for the start of the game they would make sense, in my opinion.

Agreed, I somehow mixed up some ideas in my head. I was thinking about setting those user defaults when loading a game as well, which doesn't make sense.
Those settings should be loaded when starting a game, after that the savegame is master (just like it it is now for settings)

**Comment by JonathanSteinbuch** [comment]
How about a button in the main menu to a dialogue (maybe called "configurations" or "presets") with which you can safe the current configuration? At game start it would by default load the last saved configuration but you can choose to load any other saved configuration or the standard configuration (maybe through the same dialogue).

**Comment by Spikeone** [comment]
Hm, I thought about a dialogue in the options menu, not really sure if you'd need more than 1 preset, do you?

**Comment by JonathanSteinbuch** [comment]
I was mainly thinking about the fact that to be able to open ingame settings windows you basically have to have a (fake) game open anyway. Which is why I wanted to do it with a button ingame.

I agree that it is not really necessary to allow multiple presets. However, I was thinking that you want at least a default button. And if you accidently reset everything to defaults you might want to load a previous preset again...

**Comment by Spikeone** [comment]
I understand that, but I think a user would expect such setting... in the settings? On the other hand, doing a "my current interface" snapshot is nice as well.
Not sure which is the best way to go
